server/sonar-server/src/main/java/org/sonar/server/db/DatabaseChecker.java

```diff
@@ -58,16 +59,26 @@ public void stop() {
     // nothing to do
   }
 
-  private void checkOracleDriverVersion() throws SQLException {
+  private void checkOracleVersion() throws SQLException {
     Connection connection = db.getDataSource().getConnection();
     try {
+      // check version of db
+      // See http://jira.codehaus.org/browse/SONAR-6434
+      int majorVersion = connection.getMetaData().getDatabaseMajorVersion();
+      if (majorVersion < 11) {
+        throw MessageException.of(String.format(
+          "Unsupported Oracle version: %s. Minimal required version is 11g.", connection.getMetaData().getDatabaseProductVersion()));
+      }
+
+      // check version of driver
       String driverVersion = connection.getMetaData().getDriverVersion();
       String[] parts = StringUtils.split(driverVersion, ".");
       int intVersion = Integer.parseInt(parts[0]) * 100 + Integer.parseInt(parts[1]);
       if (intVersion < 1102) {
```

Comment:
Harder as 1102 is a trick (not directly a functional version) and error message is formatted differently. This one does not worry me as much as the magic number 11 as line 68.